clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -521,8 +518,7 @@ class MarkDeviceFunction : public RecursiveASTVisitor<MarkDeviceFunction> {
         if (!CheckSYCLType(Field->getType(), Field->getSourceRange(),
                            Visited)) {
           if (SemaRef.getLangOpts().SYCLIsDevice)
-            SemaRef.SYCLDiagIfDeviceCode(Loc.getBegin(),
-                                         diag::note_sycl_used_here);
+            SemaRef.Diag(Loc.getBegin(), diag::note_sycl_used_here);
```

Comment:
Is this switchover because we already know it is emitted if we are here?

Reply:
Yeah